### PR TITLE
Adds form cleanup for user forms

### LINKED ISSUE #1653: user permissions form: use dispatchOrStoreErrors

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #1653

Some modifications to the user permissions table up for discussion. They were broken as they were to begin with.
screen shot 2017-05-08 at 6 15 30 pm
screen shot 2017-05-08 at 6 15 38 pm

### CODE REVIEW

File: src/users/user/components/PermissionsTable.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React, { PropTypes } from 'react';
+
+import { Table } from 'linode-components/tables';
+import { CheckboxCell } from 'linode-components/tables/cells';
+
+
+export default function PermissionsTable(props) {
+  const { onCellChange, parentKey, columns, objects, title } = props;
+  const headerColumn = { dataKey: 'label', label: title };
+  const allColumns = [headerColumn, ...columns.map(col => ({
```
Comment (mparke) [inline src/users/user/components/PermissionsTable.js]: do you mind breaking this into a couple lines for readability

Comment (eatonphil) [inline src/users/user/components/PermissionsTable.js]: done.

**Comment by mparke** [comment]
while you're updating styles, can you adjust the header spacing here? they should be close to their relative tables.

**Comment by eatonphil** [comment]
Ah, good call.

**Comment by mparke** [comment]
No message on user dash? Intentional?
screen shot 2017-05-09 at 4 38 31 pm

**Comment by eatonphil** [comment]
@mparke not intentional. will fix, thanks!

**Comment by eatonphil** [comment]
@mparke done.

**Comment by mparke** [comment]
Warning found, low priority follow up: https://github.com/linode/manager/issues/1744

**Pull request merged**